src/platform/posix/tcp_ip_channel.c

```diff
@@ -173,11 +184,11 @@ TaggedMessage *TcpIpChannel_receive(NetworkChannel *untyped_self) {
   return &self->output;
 }
 
-void TcpIpChannel_close(NetworkChannel *untyped_self) {
+static void TcpIpChannel_close_connection(NetworkChannel *untyped_self) {
   LF_DEBUG(NET, "Closing TCP/IP Channel");
   TcpIpChannel *self = (TcpIpChannel *)untyped_self;
 
-  if (self->server) {
+  if (self->server && self->client != 0) {
```

Comment:
This extra check is needed because in my test I sometimes just open a connection only and don't actually create a client socket. Then cleaning a client socket here creates an error.